test/core.jl

```diff
@@ -690,6 +690,18 @@ let A = [1]
     @test x == 1
 end
 
+# Pointer finalizer (issue #15408)
+let A = [1]
+    function f(to_finalize::Array{Int64,1})
+      to_finalize[1] = 0
+      nothing
+    end
+    cf = cfunction(f, Void, (Array{Int64,1},))
+    finalizer(A, cf)
```

Comment:
(in general, calling any julia runtime in a function pointer finalizer is not allowed.)